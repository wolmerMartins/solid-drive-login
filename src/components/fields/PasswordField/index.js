import React from 'react'

import FieldBase from '../FieldBase'
import InputPassword from '../../inputs/InputPassword'

const PasswordField = ({ value, onChange }) => (
  <FieldBase
    label="Senha"
    id="password-field"
  >
    <InputPassword
      value={value}
      id="password-field"
      onChange={onChange}
    />
  </FieldBase>
)

export default PasswordField
