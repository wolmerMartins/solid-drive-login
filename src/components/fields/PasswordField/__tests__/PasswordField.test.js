import React, { useState } from 'react'
import { render, fireEvent } from '@testing-library/react'

import PasswordField from '../index'

const TestComponent = () => {
  const [value, setValue] = useState('')

  const onChange = e => setValue(e.target.value)

  return <PasswordField onChange={onChange} value={value} />
}

const buildComponent = () => {
  const utils = render(<TestComponent />)
  const input = utils.getByLabelText('Senha')

  return {
    input,
    ...utils
  }
}

describe('<PasswordField />', () => {
  it('Render the component without crashing', () => {
    const { asFragment } = buildComponent()

    expect(asFragment())
      .toMatchSnapshot()
  })

  it('Types a value in the input', () => {
    const { input } = buildComponent()

    expect(input)
      .toHaveProperty('value', '')

    fireEvent.change(input, { target: { value: 'password' } })

    expect(input)
      .toHaveProperty('value', 'password')
  })
})
