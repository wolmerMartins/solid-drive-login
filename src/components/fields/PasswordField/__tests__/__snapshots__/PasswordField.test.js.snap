// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordField /> Render the component without crashing 1`] = `
<DocumentFragment>
  <fieldset
    class="FieldBase"
  >
    <label
      class="FieldBase-label"
      for="password-field"
    >
      Senha
    </label>
    <input
      class="InputBase "
      id="password-field"
      placeholder="Senha"
      type="password"
      value=""
    />
  </fieldset>
</DocumentFragment>
`;
